src/leveled_setoplexer.xrl

```diff
@@ -0,0 +1,24 @@
+Definitions.
+WhiteSpace  = ([\t\f\v\r\n\s]+)
+
+Rules.
+
+{WhiteSpace} : skip_token.
+
+\(        : {token, {'(', TokenLine}}.
+\)        : {token, {')', TokenLine}}.
+
+OR        : {token, {'UNION', TokenLine}}.
+AND       : {token, {'INTERSECT', TokenLine}}.
+NOT       : {token, {'SUBTRACT', TokenLine}}.
+UNION     : {token, {'UNION', TokenLine}}.
+INTERSECT : {token, {'INTERSECT', TokenLine}}.
+SUBTRACT  : {token, {'SUBTRACT', TokenLine}}.
+
+\$[1-8]+     : {token, {set_id, TokenLine, strip_identifier(TokenChars)}}.
```

Comment:
This feels limiting... zero and nine cannot be in the `set_id`? Do you actually mean a number not starting with zero?
`\$[1-9][0-9]*`

Reply:
In particular since you take the "nth" set_id from a list, missing the "9" is confusing if you allow "11". 
I like the map you use in filters better, then it does not really matter what representation you use. Is that possible here as well, to provide `apply_setop` with a map of `set_id` to the actual set?